### PR TITLE
Add styled-components@1.4.6 w/ npm auto-update

### LINKED ISSUE #11178: [Request] Add styled-components
**Library name:** styled-components
**Git repository url:** https://github.com/styled-components/styled-components
**npm package name or url** (if there is one): https://www.npmjs.com/package/styled-components
**License (List them all if it's multiple):** 
**Official homepage:** 
**Wanna say something? Leave message here:** 

=====================

Notes from cdnjs maintainer:
Please read the [README.md](https://github.com/cdnjs/cdnjs#cdnjs-library-repository) and [CONTRIBUTING.md](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md) document first.

We encourage you to add a library via sending pull request,
it'll be faster than just opening a request issue,
since there are tons of issues, please wait with patience,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by dakshshah96**
close #11178, cc @geelen

Pull request for issue: #11178 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/styled-components/styled-components
 * Official website (optional, not the repository): https://styled-components.com/
 * NPM package url (optional): https://www.npmjs.com/package/styled-components
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 179
   - Count of stars: 6,810
   - Count of forks: 355
 * NPM download stats (optional):
   - Downloads in the last day: 7,372
   - Downloads in the last week: 39,072
   - Downloads in the last month: 139,875

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by dakshshah96** [comment]
@pvnr0082t `styled-components.es.js` is an ES6 file. No minification services currently support minifying ES6. What do you suggest we do?

**Comment by pvnr0082t** [comment]
@dakshshah96 I think you can try to use https://github.com/PeterDaveHello/web-minify-helper. It seems that it can minify the file.

**Comment by dakshshah96** [comment]
@pvnr0082t `web-minify-helper` is unable to minify the ES6 file. Here's what I get.

screenshot 2017-05-11 09 33 55

**Comment by pvnr0082t** [comment]
@dakshshah96 Okay. Maybe we just don't minify it.

**Comment by PeterDaveHello** [comment]
No? https://github.com/PeterDaveHello/web-minify-helper/commit/a0445eb7e962329b897bf2eb0c1f6a7e3f590385

**Comment by PeterDaveHello** [comment]
My concern is that should we host it?

**Comment by dakshshah96** [comment]
@PeterDaveHello Hosting the ES6 library file is optional. The other files already include the library. ES6 can be hosted if we want users to have option of adding the ES6 edition of the library. What do you say?

**Comment by PeterDaveHello** [comment]
@dakshshah96 my concern is not just about es6, as you can see in its head, I'm not sure if "import React, { Component, createElement } from 'react';" will work, can you confirm that?

**Comment by dakshshah96** [comment]
@PeterDaveHello It should work since ES6 supports it. https://blog.jayway.com/2015/03/04/using-react-with-ecmascript-6/

**Comment by PeterDaveHello** [comment]
@dakshshah96 I mean the import part, not the syntax, as we don't host them together, will that work?

**Comment by dakshshah96** [comment]
@PeterDaveHello Yes, I think it won't work. We should not host the ES6 file then.

**Comment by PeterDaveHello** [comment]
@pvnr0082t let's use review request instead of assign.

File: ajax/libs/styled-components/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    "url": "https://github.com/styled-components/styled-components.git"
+  },
+  "license": "MIT",
+  "author": "Glen Maddern <@geelen>",
+  "npmName": "styled-components",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/*.!(*es.js)"
```
Comment (PeterDaveHello) [inline ajax/libs/styled-components/package.json]: Should be `"**/!(*.es.js)"`.

@cdnjs/intern3 didn't you guys really test it?

Comment (dakshshah96) [inline ajax/libs/styled-components/package.json]: @PeterDaveHello Done 👍

**Comment by PeterDaveHello** [comment]
@maruilian11 remember to rebase please.

**Pull request merged**